File: Find Missing Number in a sorted array.py


def missingval():
    a=[1,2,4,5,6,7,8]
    lo,hi=0,len(a)-1
    while lo<=hi:
        mid=lo+(hi-lo)//2
        if a[mid]>mid+1:
            hi=mid-1
        else:
            lo=mid+1
    return a[hi]+1

print(missingval())    

File: Find Missing Number in a sorted array.java
public int findMissingSorted(int[] nums){
int low = 0;
int high = nums.length-1;
int mid = low + (high-low)/2;

while(low < high){
if(nums[mid]-1==mid){
low = mid+1;
} else {
high = mid - 1;
}
mid = low + (high-low)/2;
}

return low;

}
